presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -163,12 +165,15 @@ public static BindableAggregationFunction generateBindableAggregationFunction(Cl
                                 genericImplementations.build()));
     }
 
-    private static AggregationImplementation parseImplementation(Class<?> aggregationDefinition, AggregationHeader header, Class<?> stateClass, Method inputFunction, Method outputFunction)
+    private static AggregationImplementation parseImplementation(Class<?> aggregationDefinition, AggregationHeader header, Class<?> stateClass, Method inputFunction, Method combineFunction, Method outputFunction)
     {
         AggregationFunction aggregationAnnotation = aggregationDefinition.getAnnotation(AggregationFunction.class);
 
+        List<ScalarImplementation.ImplementationDependency> inputDependencies = parseImplementationDependencies(inputFunction);
+        List<ScalarImplementation.ImplementationDependency> combineDependencies = parseImplementationDependencies(combineFunction);
+        List<ScalarImplementation.ImplementationDependency> outputDependencies = parseImplementationDependencies(outputFunction);
         List<LongVariableConstraint> longVariableConstraints = parseLongVariableConstraints(inputFunction);
-        List<TypeVariableConstraint> typeVariableConstraints = parseTypeVariableConstraints(inputFunction);
+        List<TypeVariableConstraint> typeVariableConstraints = parseTypeVariableConstraints(inputFunction, inputDependencies);
```

Comment:
I don't know whether any function would depend on this feature, but all 3 xDependencies should be passed to `parseTypeVariableConstraints` here (for coherence).